src/module/iotjs_module_fs.cpp

```diff
@@ -62,6 +62,20 @@ static void After(uv_fs_t* req) {
         jarg.Add(arg1);
         break;
       }
+      case UV_FS_SCANDIR:
+      {
+        int r;
+        uv_dirent_t ent;
+        uint32_t idx = 0;
+        /* make an empty javascript object */
+        JObject ret(0, (const char*)NULL);
+        while ((r = uv_fs_scandir_next(req, &ent)) != UV_EOF) {
+          ret.SetPropertyByIdx(idx, JObject(ent.name));
+          idx++;
+        }
+        jarg.Add(ret);
```

Comment:
No. I assume you are talking about #485. All current implementation is based on C++ implementaion (except for String). IMHO, I think switching to C needs more time for discussing and shaping the APIs.